pipes.Rmd

```diff
@@ -1,113 +1,120 @@
+Un pipe es una herramienta poderosa para claramente expresar una secuencia de operaciones múltiples. Hasta aquí, has venido usándolas sin saber cómo funcionan, o cuales son las alternativas. Ahora, en este capítulo, es tiempo de explorar las Pipes en más detalle. Aprenderás las alternativas a las Pipes, cuando no deberías usar las pipes y algunas herramientas relacionadas útiles. 
 
-### Prerequisites
+### Prerequisitos 
 
-The pipe, `%>%`, comes from the __magrittr__ package by Stefan Milton Bache. Packages in the tidyverse load `%>%` for you automatically, so you don't usually load magrittr explicitly.  Here, however, we're focussing on piping, and we aren't loading any other packages, so we will load it explicitly.
+Un pipe, `%>%`, viene del paquete __magrittr__  de Stefan Milton Bache. Los paquetes en el tidyverse cargan %>% automáticamente, así que no cargaras magrittr explícitamente. Aquí, sin embargo, nos enfocamos en la tubería, y no estamos cargando otros paquetes, por lo que nosotros lo cargaremos explícitamente
 
 ```{r setup, message = FALSE}
 library(magrittr)
+library(datos)
 ```
 
-## Piping alternatives
+## Alternativas a los pipes 
 
-The point of the pipe is to help you write code in a way that is easier to read and understand. To see why the pipe is so useful, we're going to explore a number of ways of writing the same code. Let's use code to tell a story about a little bunny named Foo Foo:
+El objetivo de un pipe es ayudarte a escribir código en una manera que sea más fácil de leer y entender. Para ver porque un pipe es tan útil, vamos a explorar diferentes formas de escribir el mismo código. Usemos el código para contar una historia acerca de un pequeño conejito llamado Foo Foo:
 
-> Little bunny Foo Foo  
-> Went hopping through the forest  
-> Scooping up the field mice  
-> And bopping them on the head  
+> Pequeño conejito Foo Foo  
+> Fue saltando por el bosque 
+> Recogiendo ratones del campo   
+> Y golpeándolos en la cabeza
 
-This is a popular Children's poem that is accompanied by hand actions.
+Este es un poema popular para los niños que se acompaña mediante gesticulación con las manos. 
 
-We'll start by defining an object to represent little bunny Foo Foo:
+Empezaremos por definir un objeto que representa un pequeño conejito Foo Foo:
 
 ```{r, eval = FALSE}
-foo_foo <- little_bunny()
+foo_foo <- pequeño_conejito()
 ```
 
-And we'll use a function for each key verb: `hop()`, `scoop()`, and `bop()`. Using this object and these verbs, there are (at least) four ways we could retell the story in code:
+Y usaremos una función para cada verbo: `saltar()`, `recoger()`, and `golpear()`. Usando este objeto y estos verbos, hay (al menos) cuatro maneras en las que podemos volver a contar la historia en código:
```

Comment:
- falta traducir el and del principio
- quizas me suena mas comun decir existen en lugar de hay

Reply:
Acepto